File: AINovalServer/src/main/java/com/ainovel/server/web/dto/ImportStatus.java
package com.ainovel.server.web.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * 小说导入状态DTO
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ImportStatus {

    /**
     * 状态码：PROCESSING, SAVING, INDEXING, COMPLETED, FAILED, ERROR
     */
    private String status;

    /**
     * 状态详细信息
     */
    private String message;

    /**
     * 进度百分比（可选）
     */
    private Double progress;

    public ImportStatus(String status, String message) {
        this.status = status;
        this.message = message;
        this.progress = null;
    }
}

File: AINoval/lib/models/import_status.dart
/// 小说导入状态模型
class ImportStatus {
  /// 从JSON创建实例
  factory ImportStatus.fromJson(Map<String, dynamic> json) {
    return ImportStatus(
      status: json['status'] as String,
      message: json['message'] as String,
    );
  }

  /// 创建导入状态
  ImportStatus({
    required this.status,
    required this.message,
  });

  /// 导入状态 (PROCESSING, SAVING, INDEXING, COMPLETED, FAILED, ERROR)
  final String status;

  /// 状态消息
  final String message;

  /// 转换为JSON
  Map<String, dynamic> toJson() {
    return {
      'status': status,
      'message': message,
    };
  }

  @override
  String toString() => 'ImportStatus{status: $status, message: $message}';
}
